[macOS] Expose DeviceMotionEvent / DeviceOrientationEvent on the global Window object

cdumez opened the pull request:
#### 2746089d71f266790dc9ea4a93a62472ecfd910f
<pre>
[macOS] Expose DeviceMotionEvent / DeviceOrientationEvent on the global Window object
<a href="https://bugs.webkit.org/show_bug.cgi?id=256994">https://bugs.webkit.org/show_bug.cgi?id=256994</a>

Reviewed by Brent Fulgham.

Expose DeviceMotionEvent / DeviceOrientationEvent on the global Window object
on macOS. Historically, we&apos;ve only exposed the event types on iOS since we only
fire those events on iOS. However, Chrome and Firefox expose those event types
on desktop even though they don&apos;t fire them. Exposing the event types doesn&apos;t
hurt so align with other browsers.

* LayoutTests/imported/w3c/web-platform-tests/dom/events/EventTarget-dispatchEvent-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/nodes/Document-createEvent.https-expected.txt:
* Source/WTF/wtf/PlatformEnableCocoa.h:

Canonical link: <a href="https://commits.webkit.org/264271@main">https://commits.webkit.org/264271@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/499d090dc2110bdc095c5c69da7113da23fac782

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/7198 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/7447 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/7625 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/8816 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/7418 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/7208 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/8782 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7380 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/10324 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/7325 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/8025 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/6635 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/8923 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/5376 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/6554 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/14295 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/6108 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/7017 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/6656 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/9537 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/6784 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/7134 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/5833 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/7337 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/6493 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/6463 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/1693 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1706 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/10693 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/7540 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/6877 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/1833 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebKitLegacy/mac/WebView/WebView.mm:
```diff
@@ -1557,9 +1557,6 @@ - (void)_commonInitializationWithFrameName:(NSString *)frameName groupName:(NSSt
     WebCore::provideNotification(_private->page, new WebNotificationClient(self));
 #endif
-#if ENABLE(DEVICE_ORIENTATION) && !PLATFORM(IOS_FAMILY)
-    WebCore::provideDeviceOrientationTo(*_private->page, *new WebDeviceOrientationClient(self));
```
It seems like this is the only place `WebCore::provideDeviceOrientationTo()` was used. Could we also remove the implementation of this method, since it seems to be dead code now?

cdumez replied:
Ok.

brentfulgham reviewed (approved):
Looks great!

cdumez committed:
[macOS] Expose DeviceMotionEvent / DeviceOrientationEvent on the global Window object
https://bugs.webkit.org/show_bug.cgi?id=256994

Reviewed by Brent Fulgham.

Expose DeviceMotionEvent / DeviceOrientationEvent on the global Window object
on macOS. Historically, we've only exposed the event types on iOS since we only
fire those events on iOS. However, Chrome and Firefox expose those event types
on desktop even though they don't fire them. Exposing the event types doesn't
hurt so align with other browsers.

* LayoutTests/imported/w3c/web-platform-tests/dom/events/EventTarget-dispatchEvent-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/nodes/Document-createEvent.https-expected.txt:
* Source/WTF/wtf/PlatformEnableCocoa.h:

Canonical link: https://commits.webkit.org/264271@main
